piksi_tools/fileio.py

```diff
@@ -393,11 +395,11 @@ def print_dir_listing(files):
 
     Parameters
     ----------
-    files : [str]
+    files : [bytes]
         List of file names in the directory.
     """
     for f in files:
-        print(f)
+        print(f.decode('ascii', 'replace'))
```

Comment:
I think we can safely make this `'utf8'` instead of `'ascii'` to add a little bit of future proofing.

```suggestion
        print(f.decode('utf8', 'replace'))
```

Reply:
Pondered a bit more on this non-utf-8 caller terminal situation and came up with a slightly complex solution, added as a separate commit: 
1. first decode device utf-8 bytes into python's internal string format
2. then encode that string to stdout's encoding (bytes) with error replacement
3. then decode back into python's string format. This string can be printed out with the print command

This has the advantage that no gibberish is printed out even if the user's system does not support utf-8 as long as locale envs are configured properly. If unicode characters from the device can be represented in the user's encoding, they will be, otherwise they will be replaced by question marks.